Get only the facilities that we have diagnosis data for

In [5]:
import pandas as pd

In [11]:
diag = pd.read_csv("../../data/external/health/medicare-claims-weekly.csv")
facilities = pd.read_csv('../../data/external/health/UT_Facilities.csv')

In [12]:
diag.merge(facilities, on='PRVDR_NUM', how='left')

patient_count related_diagnosis  claim_week  claim_year PRVDR_NUM  \
0                12      A_AB_COPD_DY           1        2018    460006   
1                12      A_AB_COPD_DY           1        2020    460001   
2                36      A_AB_COPD_DY           1        2022    460009   
3                53      A_AB_COPD_DY           1        2021    460009   
4                16      A_AB_COPD_DY           1        2021    460047   
...             ...               ...         ...         ...       ...   
2360             12      A_AB_COPD_DY          53        2020    460019   
2361             12      A_AB_COPD_DY          53        2020    460021   
2362             11      A_AB_COPD_DY          53        2020    46T003   
2363             18      A_AB_COPD_DY          53        2020    460004   
2364             36      A_AB_COPD_DY          53        2020    460009   

      PRVDR_STATE_CD       CITY_NAME  \
0                 46  SALT LAKE CITY   
1                 46           PROVO   
2                 46  SALT LAKE CITY   
3                 46  SALT LAKE CITY   
4                 46  SALT LAKE CITY   
...              ...             ...   
2360              46       ROOSEVELT   
2361              46       ST GEORGE   
2362              46             NaN   
2363              46           OGDEN   
2364              46  SALT LAKE CITY   

                                      FAC_NAME STATE_CD   ZIP_CD  
0                                 LDS HOSPITAL       UT  84143.0  
1                         UTAH VALLEY HOSPITAL       UT  84604.0  
2     UNIVERSITY OF UTAH HOSPITALS AND CLINICS       UT  84132.0  
3     UNIVERSITY OF UTAH HOSPITALS AND CLINICS       UT  84132.0  
4                           ST MARK'S HOSPITAL       UT  84124.0  
...                                        ...      ...      ...  
2360               UINTAH BASIN MEDICAL CENTER       UT  84066.0  
2361             DIXIE REGIONAL MEDICAL CENTER       UT  84790.0  
2362                                       NaN      NaN      NaN  
2363                        MCKAY DEE HOSPITAL       UT  84403.0  
2364  UNIVERSITY OF UTAH HOSPITALS AND CLINICS       UT  84132.0  

[2365 rows x 10 columns]

In [13]:
relevant_fac = facilities[facilities['PRVDR_NUM'].isin(diag['PRVDR_NUM'].unique())]
relevant_fac

CITY_NAME                                           FAC_NAME  \
0             PROVO                               UTAH VALLEY HOSPITAL   
1    SALT LAKE CITY                  SALT LAKE REGIONAL MEDICAL CENTER   
2             OGDEN                                 MCKAY DEE HOSPITAL   
3             OGDEN                      OGDEN REGIONAL MEDICAL CENTER   
4    SALT LAKE CITY                                       LDS HOSPITAL   
5        CEDAR CITY                                CEDAR CITY HOSPITAL   
7    SALT LAKE CITY           UNIVERSITY OF UTAH HOSPITALS AND CLINICS   
8            MURRAY                       INTERMOUNTAIN MEDICAL CENTER   
9             PRICE                                CASTLEVIEW HOSPITAL   
11           PAYSON                             MOUNTAIN VIEW HOSPITAL   
12           TOOELE                       MOUNTAIN WEST MEDICAL CENTER   
13            LOGAN                            LOGAN REGIONAL HOSPITAL   
15     BRIGHAM CITY                    BRIGHAM CITY COMMUNITY HOSPITAL   
17        ROOSEVELT                        UINTAH BASIN MEDICAL CENTER   
19        ST GEORGE                      DIXIE REGIONAL MEDICAL CENTER   
21    AMERICAN FORK                             AMERICAN FORK HOSPITAL   
24        RICHFIELD                             SEVIER VALLEY HOSPITAL   
28           VERNAL                     ASHLEY REGIONAL MEDICAL CENTER   
34        TREMONTON                         BEAR RIVER VALLEY HOSPITAL   
36           LAYTON                  DAVIS HOSPITAL AND MEDICAL CENTER   
37        BOUNTIFUL                                  LAKEVIEW HOSPITAL   
39            SANDY                                 ALTA VIEW HOSPITAL   
41   SALT LAKE CITY                                 ST MARK'S HOSPITAL   
43           MURRAY                  THE ORTHOPEDIC SPECIALTY HOSPITAL   
46      WEST JORDAN                       JORDAN VALLEY MEDICAL CENTER   
47             OREM                       TIMPANOGOS REGIONAL HOSPITAL   
49      NORTH LOGAN                             CACHE VALLEY  HOSPITAL   
52        PARK CITY                                 PARK CITY HOSPITAL   
53         RIVERTON                                  RIVERTON HOSPITAL   
55           DRAPER                                 LONE PEAK HOSPITAL   
56           LAYTON                                    LAYTON HOSPITAL   
59            DELTA                           DELTA COMMUNITY HOSPITAL   
60         FILLMORE                        FILLMORE COMMUNITY HOSPITAL   
61             MOAB                             MOAB REGIONAL HOSPITAL   
62   MOUNT PLEASANT                      SANPETE VALLEY HOSPITAL - CAH   
63            NEPHI                CENTRAL VALLEY MEDICAL CENTER - CAH   
65         GUNNISON                           GUNNISON VALLEY HOSPITAL   
66       HEBER CITY                              HEBER VALLEY HOSPITAL   
67       MONTICELLO                                  SAN JUAN HOSPITAL   
68            KANAB                               KANE COUNTY HOSPITAL   
69         BLANDING                             BLUE MOUNTAIN HOSPITAL   
70        PANGUITCH                         GARFIELD MEMORIAL HOSPITAL   
71           BEAVER                             BEAVER VALLEY HOSPITAL   
197     EAST CARBON                 CARBON MEDICAL SERVICE ASSOCIATION   
198        BICKNELL                     WAYNE COMMUNITY HEALTH CENTERS   
211           OGDEN          M C H C - MIDTOWN COMMUNITY HEALTH CENTER   
213  SALT LAKE CITY                       WASATCH HOMELESS HEALTH CARE   
215       ST GEORGE  S W U C H C - SOUTHWEST UT COMMUNITY HEALTH CE...   
229        BLANDING   BLANDING FAMILY PRACTICE COMMUNITY HEALTH CENTER   
233          VERNAL          MOUNTAINLANDS COMMUNITY HEALTH CENTER INC   
289           PROVO                        UTAH VALLEY DIALYSIS CENTER   
304           OGDEN                           LIBERTY DIALYSIS - OGDEN   
307          PAYSON  UNIVERSITY OF UTAH PAYSON REGIONAL DIALYSIS CE...   
312      CEDAR

Looks like we have some facilities in the diagnosis data that aren't in the facilities data. ID those below

In [27]:
diag[~diag['PRVDR_NUM'].isin(facilities['PRVDR_NUM'])]

patient_count related_diagnosis  claim_week  claim_year PRVDR_NUM  \
21               12      A_AB_COPD_DY           1        2018    461835   
246              12      A_AB_COPD_DY           6        2020    461835   
329              12      A_AB_COPD_DY           7        2022    463994   
891              12      A_AB_COPD_DY          20        2022    460062   
931              12            Emesis          21        2017    46T051   
2309             12      A_AB_COPD_DY          51        2022    461835   
2362             11      A_AB_COPD_DY          53        2020    46T003   

      PRVDR_STATE_CD  
21                46  
246               46  
329               46  
891               46  
931               46  
2309              46  
2362              46

With Korin's guidance, we won't worry about the providers that have a T in their `PRVDR_NUM` if we can't easily find a match for them.

Notes:
* 461835 can't find
* 463994 is Roosevelt, UT outside of our region of interest
* 460062 is Spanish Fork, UT which we do want to present data for

Order the facilities by zip

In [15]:
relevant_fac = relevant_fac.sort_values('ZIP_CD').reset_index()
relevant_fac.head(60)

index       CITY_NAME                                           FAC_NAME  \
0      21   AMERICAN FORK                             AMERICAN FORK HOSPITAL   
1      37       BOUNTIFUL                                  LAKEVIEW HOSPITAL   
2      55          DRAPER                                 LONE PEAK HOSPITAL   
3      66      HEBER CITY                              HEBER VALLEY HOSPITAL   
4      56          LAYTON                                    LAYTON HOSPITAL   
5      36          LAYTON                  DAVIS HOSPITAL AND MEDICAL CENTER   
6      47            OREM                       TIMPANOGOS REGIONAL HOSPITAL   
7      52       PARK CITY                                 PARK CITY HOSPITAL   
8      53        RIVERTON                                  RIVERTON HOSPITAL   
9      17       ROOSEVELT                        UINTAH BASIN MEDICAL CENTER   
10     12          TOOELE                       MOUNTAIN WEST MEDICAL CENTER   
11     28          VERNAL                     ASHLEY REGIONAL MEDICAL CENTER   
12    233          VERNAL          MOUNTAINLANDS COMMUNITY HEALTH CENTER INC   
13     46     WEST JORDAN                       JORDAN VALLEY MEDICAL CENTER   
14     39           SANDY                                 ALTA VIEW HOSPITAL   
15    213  SALT LAKE CITY                       WASATCH HOMELESS HEALTH CARE   
16      1  SALT LAKE CITY                  SALT LAKE REGIONAL MEDICAL CENTER   
17      8          MURRAY                       INTERMOUNTAIN MEDICAL CENTER   
18     43          MURRAY                  THE ORTHOPEDIC SPECIALTY HOSPITAL   
19    384        HOLLADAY                          PRECISION HEALTH SERVICES   
20     41  SALT LAKE CITY                                 ST MARK'S HOSPITAL   
21      7  SALT LAKE CITY           UNIVERSITY OF UTAH HOSPITALS AND CLINICS   
22      4  SALT LAKE CITY                                       LDS HOSPITAL   
23     15    BRIGHAM CITY                    BRIGHAM CITY COMMUNITY HOSPITAL   
24     34       TREMONTON                         BEAR RIVER VALLEY HOSPITAL   
25     13           LOGAN                            LOGAN REGIONAL HOSPITAL   
26     49     NORTH LOGAN                             CACHE VALLEY  HOSPITAL   
27    211           OGDEN          M C H C - MIDTOWN COMMUNITY HEALTH CENTER   
28    304           OGDEN                           LIBERTY DIALYSIS - OGDEN   
29      2           OGDEN                                 MCKAY DEE HOSPITAL   
30      3           OGDEN                      OGDEN REGIONAL MEDICAL CENTER   
31      9           PRICE                                CASTLEVIEW HOSPITAL   
32    229        BLANDING   BLANDING FAMILY PRACTICE COMMUNITY HEALTH CENTER   
33     69        BLANDING                             BLUE MOUNTAIN HOSPITAL   
34    358     CASTLE DALE                               EMERY MEDICAL CENTER   
35    197     EAST CARBON                 CARBON MEDICAL SERVICE ASSOCIATION   
36     61            MOAB                             MOAB REGIONAL HOSPITAL   
37     67      MONTICELLO                                  SAN JUAN HOSPITAL   
38    289           PROVO                        UTAH VALLEY DIALYSIS CENTER   
39      0           PROVO                               UTAH VALLEY HOSPITAL   
40     59           DELTA                           DELTA COMMUNITY HOSPITAL   
41     60        FILLMORE                        FILLMORE COMMUNITY HOSPITAL   
42     65        GUNNISON                           GUNNISON VALLEY HOSPITAL   
43     62  MOUNT PLEASANT                      SANPETE VALLEY HOSPITAL - CAH   
44     63           NEPHI                CENTRAL VALLEY MEDICAL CENTER - CAH   
45    307          PAYSON  UNIVERSITY OF UTAH PAYSON REGIONAL DIALYSIS CE...   
46     11          PAYSON                             MOUNTAIN VIEW HOSPITAL   
47    315           SALEM                                   MT NEBO DIALYSIS   
48    341       RICHFIELD                      MOUNTAIN UTAH FAMILY MEDICINE   
49  

In [17]:
lat = pd.Series([
    40.37985836142577, # american fork hospital
    40.88616998241446, # lakeview hospital
    #40.37631202166172, # lone peak
    40.53541691211239, # lone peak hospital
    40.490005260306944, # herber valley hospital
    41.0522961898919, # layton hospital
    41.08971067876015, # davis hospital and medical center
    40.31337525200588, # timpanogos regional hospital
    40.68816454926236, # PARK CITY HOSPITAL
    #40.30287579906333, # OREM COMMUNITY HOSPITAL
    40.52056861385218, # RIVERTON HOSPITAL
    40.3041728978642, # UINTAH BASIN MEDICAL CENTER
    40.566612419975996, # MOUNTAIN WEST MEDICAL CENTER
    40.45768487648884, # ASHLEY REGIONAL MEDICAL CENTER
    40.226793832303, # MOUNTAINLANDS COMMUNITY HEALTH CENTER INC
    40.58995318156501,  # JORDAN VALLEY MEDICAL CENTER
    #40.60479195217989, # ENCOMPASS HEALTH REHABILITATION HOSPITAL OF UTAH
    40.57596794664961, # ALTA VIEW HOSPITAL
    40.76102295807381, # WASATCH HOMELESS HEALTH CARE
    40.76793115669842, # SALT LAKE REGIONAL MEDICAL CENTER
    40.65927003391647, # INTERMOUNTAIN MEDICAL CENTER
    40.64425586098879, # THE ORTHOPEDIC SPECIALTY HOSPITAL
    40.630164266957344, # PRECISION HEALTH SERVICES
    40.686112520582434, # ST MARK'S HOSPITAL
    40.77131159459345, # UNIVERSITY OF UTAH HOSPITALS AND CLINICS
    40.77842280310886, # LDS HOSPITAL
    41.49120184606022, # BRIGHAM CITY COMMUNITY HOSPITAL
    41.724381454046664, # BEAR RIVER VALLEY HOSPITAL
    41.75571756402917, # LOGAN REGIONAL HOSPITAL
    41.775635562246414, # CACHE VALLEY  HOSPITAL
    41.22585741197675, # M C H C - MIDTOWN COMMUNITY HEALTH CENTER
    41.17628822313022, # LIBERTY DIALYSIS OGDEN
    41.18368362589415, # MCKAY DEE HOSPITAL
    #41.157139359750225, # NORTHERN UTAH REHABILITATION HOSPITAL
    #41.16238990830809, # BONNEVILLE DIALYSIS CENTER
    41.16426445109952, # OGDEN REGIONAL MEDICAL CENTER
    #41.30802287753525, # PLEASANT VIEW DIALYSIS CENTER
    39.60515821068704, # CASTLEVIEW HOSPITAL
    37.61210614718201, # BLANDING FAMILY PRACTICE COMMUNITY HEALTH CENTER
    37.612873633606505, # BLUE MOUNTAIN HOSPITAL
    39.21236960773042, # EMERY MEDICAL CENTER
    39.5502350491628, # CARBON MEDICAL SERVICE ASSOCIATION
    38.575211157918595, # MOAB REGIONAL HOSPITAL
    #37.259996847577526, # MONTEZUMA CREEK COMMUNITY HEALTH CENTER
    #37.87446948469256, # SAN JUAN CLINIC
    37.87444486471641, # SAN JUAN HOSPITAL
    40.24856765104494, # UTAH VALLEY DIALYSIS CENTER
    40.247444226954116, # UTAH VALLEY HOSPITAL
    39.34998116828826, # DELTA COMMUNITY HOSPITAL
    38.954573623492514, # FILLMORE COMMUNITY HOSPITAL
    39.15620291623618, # GUNNISON VALLEY HOSPITAL
    39.53143169763917, # SANPETE VALLEY HOSPITAL - CAH
    39.7314994656203, # CENTRAL VALLEY MEDICAL CENTER - CAH
    40.04349252717398, # UNIVERSITY OF UTAH PAYSON REGIONAL DIALYSIS CENTER
    40.04405303694899, # MOUNTAIN VIEW HOSPITAL
    40.08431856988072, # MT NEBO DIALYSIS
    38.78049261704714, # MOUNTAIN UTAH FAMILY MEDICINE
    38.78260028891253, # SEVIER VALLEY HOSPITAL
    38.28898085075151, # BEAVER VALLEY HOSPITAL
    38.28871118256148, # BEAVER MEDICAL CLINIC
    38.338862595849605, # WAYNE COMMUNITY HEALTH CENTERS
    37.700520655989905, # FRESENIUS MEDICAL CARE CEDAR CITY LLC
    37.69957419307554, # CEDAR CITY HOSPITAL
    37.054344672175745, # KANE COUNTY HOSPITAL
    37.82635093934383, # GARFIELD MEMORIAL HOSPITAL
    #37.826427030765586, # GARFIELD MEMORIAL CLINIC
    #37.097341982866155, # LIBERTY DIALYSIS - ST GEORGE
    37.11099523090387, # S W U C H C - SOUTHWEST UT COMMUNITY HEALTH CENTER
    37.09882346306926 # DIXIE REGIONAL MEDICAL CENTER
])

In [18]:
lon = pd.Series([
     -111.76697077644893, # american fork hospital
    -111.86872416293062, # lakeview hospital
    #-111.76727465029992, # lone peak
    -111.88925100119239, # lone peak hospital
    -111.40616824760788, # herber valley hospital
    -111.96877469175783, # layton hospital
    -111.99467384388011, # davis hospital and medical center
    -111.7139662513146, # timpangogos regional hospital
    #-111.70822016158368, # OREM COMMUNITY HOSPITAL
    -111.46948059559428, # PARK CITY HOSPITAL
    -111.9802857417843, # RIVERTON HOSPITAL
    -109.99668120126964, # UINTAH BASIN MEDICAL CENTER
    -112.2978662573994, # MOUNTAIN WEST MEDICAL CENTER
    -109.53163679548543, # ASHLEY REGIONAL MEDICAL CENTER
    -111.64334256501294, # MOUNTAINLANDS COMMUNITY HEALTH CENTER INC
    -111.9753606446494, # JORDAN VALLEY MEDICAL CENTER
    #-111.85471205699126, # ENCOMPASS HEALTH REHABILITATION HOSPITAL OF UTAH
    -111.85413759178151, # ALTA VIEW HOSPITAL
    -111.90296273570496, # WASATCH HOMELESS HEALTH CARE
    -111.86156755074184, # SALT LAKE REGIONAL MEDICAL CENTER
    -111.8911506163486, # INTERMOUNTAIN MEDICAL CENTER
    -111.8837000362312, # THE ORTHOPEDIC SPECIALTY HOSPITAL
    -111.93831712558385, # PRECISION HEALTH SERVICES
    -111.85721089362508, # ST MARK'S HOSPITAL
    -111.83624924817167, # UNIVERSITY OF UTAH HOSPITALS AND CLINICS
    -111.88014380351014, # LDS HOSPITAL
    -112.02598068378951, # BRIGHAM CITY COMMUNITY HOSPITAL
    -112.1826187819288, # BEAR RIVER VALLEY HOSPITAL
    -111.82215573590031, # LOGAN REGIONAL HOSPITAL
    -111.82096426898319, # CACHE VALLEY  HOSPITAL
    -111.96714837822132, # M C H C - MIDTOWN COMMUNITY HEALTH CENTER
    -111.94744873295866, # LIBERTY DIALYSIS - OGDEN
    #-111.9411784420278, # NORTHERN UTAH REHABILITATION HOSPITAL
    -111.9536113274326, # MCKAY DEE HOSPITAL
    #-111.96932025127252, # BONNEVILLE DIALYSIS CENTER
    -111.9689487647489, # OGDEN REGIONAL MEDICAL CENTER
    #-112.00825796142924, # PLEASANT VIEW DIALYSIS CENTER
    -110.83328258628212, # CASTLEVIEW HOSPITAL
    -109.4841581879023, # BLANDING FAMILY PRACTICE COMMUNITY HEALTH CENTER
    -109.48268532583184, # BLUE MOUNTAIN HOSPITAL
    -111.0212679630126, # EMERY MEDICAL CENTER
    -110.39345516963279, # CARBON MEDICAL SERVICE ASSOCIATION
    -109.55921918448252, # MOAB REGIONAL HOSPITAL
    #-109.30390343611786, # MONTEZUMA CREEK COMMUNITY HEALTH CENTER
    #-109.34981898604534, # SAN JUAN CLINIC
    -109.35017120513693, # SAN JUAN HOSPITAL
    -111.66843196280259, # UTAH VALLEY DIALYSIS CENTER
    -111.66573682063333, # UTAH VALLEY HOSPITAL
    -112.56111189680894, # DELTA COMMUNITY HOSPITAL
    -112.34040747651811, # FILLMORE COMMUNITY HOSPITAL
    -111.81647201513941, # GUNNISON VALLEY HOSPITAL
    -111.46102513231249, # SANPETE VALLEY HOSPITAL - CAH
    -111.83792824949417, # CENTRAL VALLEY MEDICAL CENTER - CAH
    -111.71470912900799, # UNIVERSITY OF UTAH PAYSON REGIONAL DIALYSIS CENTER
    -111.71448099180772, # MOUNTAIN VIEW HOSPITAL
    -111.6846310030603, # MT NEBO DIALYSIS
    -112.08554877837547, # MOUNTAIN UTAH FAMILY MEDICINE
    -112.08404550906, # SEVIER VALLEY HOSPITAL
    -112.64381270207824, # BEAVER VALLEY HOSPITAL
    -112.643765006956, # BEAVER MEDICAL CLINIC
    -111.55011025665439, # WAYNE COMMUNITY HEALTH CENTERS
    -113.06330427978851, # FRESENIUS MEDICAL CARE CEDAR CITY LLC
    -113.06660102630221, # CEDAR CITY HOSPITAL
    -112.5296166514696, # KANE COUNTY HOSPITAL
    -112.4270901075313, # GARFIELD MEMORIAL HOSPITAL
    #-112.42719403451795, # GARFIELD MEMORIAL CLINIC
    #-113.56278287475632, # LIBERTY DIALYSIS - ST GEORGE
    -113.58095054524522, # S W U C H C - SOUTHWEST UT COMMUNITY HEALTH CENTER
    -113.57575022018614 # DIXIE REGIONAL MEDICAL CENTER
])

In [19]:
assert(len(lat) == len(lon))

In [20]:
relevant_fac['lat'] = lat
relevant_fac['lon'] = lon

In [21]:
relevant_fac

index       CITY_NAME                                           FAC_NAME  \
0      21   AMERICAN FORK                             AMERICAN FORK HOSPITAL   
1      37       BOUNTIFUL                                  LAKEVIEW HOSPITAL   
2      55          DRAPER                                 LONE PEAK HOSPITAL   
3      66      HEBER CITY                              HEBER VALLEY HOSPITAL   
4      56          LAYTON                                    LAYTON HOSPITAL   
5      36          LAYTON                  DAVIS HOSPITAL AND MEDICAL CENTER   
6      47            OREM                       TIMPANOGOS REGIONAL HOSPITAL   
7      52       PARK CITY                                 PARK CITY HOSPITAL   
8      53        RIVERTON                                  RIVERTON HOSPITAL   
9      17       ROOSEVELT                        UINTAH BASIN MEDICAL CENTER   
10     12          TOOELE                       MOUNTAIN WEST MEDICAL CENTER   
11     28          VERNAL                     ASHLEY REGIONAL MEDICAL CENTER   
12    233          VERNAL          MOUNTAINLANDS COMMUNITY HEALTH CENTER INC   
13     46     WEST JORDAN                       JORDAN VALLEY MEDICAL CENTER   
14     39           SANDY                                 ALTA VIEW HOSPITAL   
15    213  SALT LAKE CITY                       WASATCH HOMELESS HEALTH CARE   
16      1  SALT LAKE CITY                  SALT LAKE REGIONAL MEDICAL CENTER   
17      8          MURRAY                       INTERMOUNTAIN MEDICAL CENTER   
18     43          MURRAY                  THE ORTHOPEDIC SPECIALTY HOSPITAL   
19    384        HOLLADAY                          PRECISION HEALTH SERVICES   
20     41  SALT LAKE CITY                                 ST MARK'S HOSPITAL   
21      7  SALT LAKE CITY           UNIVERSITY OF UTAH HOSPITALS AND CLINICS   
22      4  SALT LAKE CITY                                       LDS HOSPITAL   
23     15    BRIGHAM CITY                    BRIGHAM CITY COMMUNITY HOSPITAL   
24     34       TREMONTON                         BEAR RIVER VALLEY HOSPITAL   
25     13           LOGAN                            LOGAN REGIONAL HOSPITAL   
26     49     NORTH LOGAN                             CACHE VALLEY  HOSPITAL   
27    211           OGDEN          M C H C - MIDTOWN COMMUNITY HEALTH CENTER   
28    304           OGDEN                           LIBERTY DIALYSIS - OGDEN   
29      2           OGDEN                                 MCKAY DEE HOSPITAL   
30      3           OGDEN                      OGDEN REGIONAL MEDICAL CENTER   
31      9           PRICE                                CASTLEVIEW HOSPITAL   
32    229        BLANDING   BLANDING FAMILY PRACTICE COMMUNITY HEALTH CENTER   
33     69        BLANDING                             BLUE MOUNTAIN HOSPITAL   
34    358     CASTLE DALE                               EMERY MEDICAL CENTER   
35    197     EAST CARBON                 CARBON MEDICAL SERVICE ASSOCIATION   
36     61            MOAB                             MOAB REGIONAL HOSPITAL   
37     67      MONTICELLO                                  SAN JUAN HOSPITAL   
38    289           PROVO                        UTAH VALLEY DIALYSIS CENTER   
39      0           PROVO                               UTAH VALLEY HOSPITAL   
40     59           DELTA                           DELTA COMMUNITY HOSPITAL   
41     60        FILLMORE                        FILLMORE COMMUNITY HOSPITAL   
42     65        GUNNISON                           GUNNISON VALLEY HOSPITAL   
43     62  MOUNT PLEASANT                      SANPETE VALLEY HOSPITAL - CAH   
44     63           NEPHI                CENTRAL VALLEY MEDICAL CENTER - CAH   
45    307          PAYSON  UNIVERSITY OF UTAH PAYSON REGIONAL DIALYSIS CE...   
46     11          PAYSON                             MOUNTAIN VIEW HOSPITAL   
47    315           SALEM                                   MT NEBO DIALYSIS   
48    341       RICHFIELD                      MOUNTAIN UTAH FAMILY MEDICINE   
49  

Add back in the Spanish Fork facility we couldn't find earlier

In [40]:
diag[diag['PRVDR_NUM'].astype("string") == "460062"]

patient_count related_diagnosis  claim_week  claim_year PRVDR_NUM  \
891             12      A_AB_COPD_DY          20        2022    460062   

     PRVDR_STATE_CD  
891              46

In [22]:
save = relevant_fac.iloc[ :, 1:]

In [42]:
save.loc[len(save.index)] = ['SPANISH FORK', 'SPANISH FORK HOSPITAL', '460062', 'UT', 84660, 40.130694887524236, -111.64088972382105] 

In [44]:
save.to_csv('../../data/processed/facility_locations_latlon.csv', index=False)